src/main/scala/com.snowplowanalytics.iglu/ctl/commands/TableCheck.scala

```diff
@@ -34,216 +32,291 @@ import com.snowplowanalytics.iglu.ctl.{Command, Common, Failing, File, Server, S
-    for {
-      resolverJson <- EitherT(File.readFile(resolverPath).map(_.flatMap(_.asJson)))
-      resolver <- EitherT(Resolver.parse[IO](resolverJson.content))
-        .leftMap(e => Error.ConfigParseError(s"Resolver can not created: $e"))
-      schemaKeyList <- EitherT(resolver.listSchemas(schemaKey.vendor, schemaKey.name, schemaKey.version.model))
-        .leftMap(e => Error.ServiceError(s"Error while lookup for schema key list: ${(e: ClientError).asJson.noSpaces}"))
-      schemasNel <- EitherT.fromOption[IO](
-        NonEmptyList.fromList(schemaKeyList.schemas), Error.ServiceError(s"Error while lookup for schema family: ${schemaKey.vendor}-${schemaKey.name}-${schemaKey.version.model}"))
-      schemas <- schemasNel.traverse(schemaKey => for {
-        schemaJson <- EitherT(resolver.lookupSchema(schemaKey))
-          .leftMap(e => Error.ServiceError(s"Error while lookup for schema: ${(e: ClientError).asJson.noSpaces}"))
-        schema <- EitherT.fromEither[IO](parseSchema(schemaJson))
-      } yield schema
-      )
-    } yield schemas
-
-  def parseSchema(schema: Json): Either[Common.Error, IgluSchema] =
-    SelfDescribingSchema.parse(schema)
-      .leftMap(e => Common.Error.Message(s"Error while parsing schema jsons to Schema object, $e"))
-      .flatMap { s =>
-        Schema.parse(s.schema)
-          .map(e => SelfDescribingSchema(s.self, e))
-          .toRight(Common.Error.Message(s"Error while parsing schema jsons to Schema object"))
+  private def groupSchemasToFamilies(schemas: NonEmptyList[SelfDescribingSchema[Schema]]): List[NonEmptyList[SelfDescribingSchema[Schema]]] = {
+    schemas
+      .groupBy { schema =>
+        (schema.self.schemaKey.vendor, schema.self.schemaKey.name, schema.self.schemaKey.version.model)
       }
+      .values
+      .toList
+      .map(_.sortBy(_.self.schemaKey))
+  }
 
 
-  /**
-    * Compares existing and expected columns whether they are same or not.
-    */
-  private[ctl] def checkColumns(schemaKey: SchemaKey, existingColumns: List[Column], expectedColumns: List[String]): Result =
-    existingColumns match {
-      case Nil => TableNotDeployed(schemaKey)
-      case _ =>
-        expectedColumns.zipAll(existingColumns.map(_.columnName), "", "").foldLeft(List.empty[(String, String)]) {
-          case (acc, (expected, existing)) =>
-            if (expected != existing) (existing, expected) :: acc else acc
-        } match {
-          case Nil => TableMatched(schemaKey)
-          case _ => TableUnmatched(schemaKey, existingColumns, expectedColumns)
-        }
+  private def parseSchema(schema: SelfDescribingSchema[Json]): Either[Common.Error, IgluSchema] = {
+    Schema.parse(schema.schema)
+      .map(e => SelfDescribingSchema(schema.self, e))
+      .toRight(Common.Error.Message(s"Error while parsing schema jsons to Schema object"))
+  }
+
+  private def mapDdlType(ddlColumnType: ShredModelEntry.ColumnType): Column.Type = {
```

Comment:
Is there an issue of using the `ShredModelEntry.ColumnType` directly? It would avoid this mapping.